### DISCUSSION TITLE
Trouble Building in Unreal 5.3.2

### DISCUSSION THREAD

**Discussion by stephenbdavies** [JSBSim-Team/jsbsim#1189] [Q&A]
Hi all, hoping someone can help me get up and running here. I'm following along to the README in in the UnrealEngine folder. I'm successfully able to get to step 4. Double clicking on the .uproject gives me the error that I should try building manually. When I right click and select "Generate Visual Studio Project Files" I get a "failed to generate project files" error and VS won't open.

Attaching the logs. I noticed a reference to the BuildException in Log_GPF which seems to me to be a conflict between my version of VS 2022 and the 2019 toolkit... but I don't know how to solve that (if it's even the case).

Before I had tried this with 5.5 and 5.4, but then after browsing the discussions noticed that it's only 5.0-5.3 that's supported. Let me know if it's worthwhile to go back to an earlier version and try that. I WAS able to get further in one of my earlier attempts in 5.5 or 5.4 but then the build failed after opening the VS solution. That may have been an attempt to not direct to the 2019 toolkit though, and from what I remember from the output window, why it failed.

Any help is appreciated - thank you!

[Log_GPF.txt](https://github.com/user-attachments/files/17808424/Log_GPF.txt)
[UnrealVersionSelector-2024.11.18-21.05.47.log](https://github.com/user-attachments/files/17808425/UnrealVersionSelector-2024.11.18-21.05.47.log)

**Comment by gallonmate** [comment]
Did you happen to do the step quoted below? If so, then delete that file and skip that step. Also click and run CleanProject.bat to clean up the bad build files. Then right click on Uproject to now generate files. If the project works, please let me know so that we can update the ReadMe file, as this step might be outdated and only useful for UE5.0
>     Force Unreal Build tool to produce VS2019 projects
> 
>     Create a file named "BuildConfiguration.xml" in %APPDATA%\Unreal Engine\UnrealBuildTool
>     (ex: C:\Users\JohnDoe\AppData\Roaming\Unreal Engine\UnrealBuildTool)
>     Edit the file so that he looks like this one with a CompilerVersion block :

If the above was not the problem:

-Verify you have MSVC 14.34 or later installed. You can open Visual Studio Installer, click modify for VS2022, go to individual components tab, then keyboard search MSVC, scroll down and see which one you have checked off as already installed. Also I would double-check that all the recommended components are installed for usage with regular Unreal Engine, per the standard documentation from Epic.

Then click/run the CleanProject.bat if you haven't already to get rid of bad build files. And right click Uproject to generate files again.

**Reply by stephenbdavies** [reply]
Hey, @gallonmate - thanks for the response. I did initially do the step where I added the compiler version block to the BuildConfiguration.xml file. This time I deleted the file and skipped that step, but no luck, unfortunately. Same result, and the logs appear identical.

I opened VS Installer, and when I search for MSVC under individual components installed, I have installed/checked off:

- C++ Build Insights
- MSVC v142 - VS 2019 C++ ARM64 build tools (v14.29-16.11)
- MSVC v142 - VS 2019 C++ ARM64 Spectre-mitigated libs (v14.29-16.11)
- MSVC v142 - VS 2019 C++ x64/x86 build tools (v14.29-16.11)
- MSVC v143 - VS 2022 C++ x64/x86 build tools (Latest)
- MSVC v143 - VS 2022 C++ x64/x86 build tools (v14.38-17.8)

I ran through the Unreal pages linked in the README and ran through everything there. When installing the UnrealVS Extensions I did it for both versions. For 2022 I got the message "Already installed on all applicable targets." For 2019 I got the error "This extension is not installable on any currently installed products" with the attached log, but I'm not sure if that's an issue if I'm using 2022 with the 2019 build tools. It does show under extensions > installed in VS when I open it.

After that I ran CleanProject.bat, removed the BuildConfiguration file, and tried again. Then I rebuilt JSBSimForUnreal.sln in the root folder. Not sure if I should do that for each of these tries, when I ran it the output stated "0 succeeded, 0 failed, 1 up-to-date, 0 skipped," so this step might've been redundant. I tried to generate the VS files off of UEReferenceApp.uproject again, but this gave me the same errors and logs.

I hope I'm not making a dumb mistake - I'm new to both Unreal and Visual Studio.

**Reply by gallonmate** [reply]
No you're not making a mistake. This is just how things are from time to time with getting development environments setup. There are a lot of the same issues having been reported from users of Unreal Engine in general. After I have skimmed through various forum threads of the same issue, it seems I would try installing the actual MSVC 14.34 version, which would be listed as - MSVC v143 - VS 2002 C++ x64/86 build tools (v14.34-17.4)

If that doesn't work, I would google the phrase "Microsoft platform targets must be compiled with Visual Studio 2022 17.4 (MSVC 14.34.x) or later for the installed engine." The results will show mostly Unreal Engine discussions and you can work your way through the various solutions/fixes people have found.

**Comment by stephenbdavies** [comment]
I ended up trying some of the things listed in various discussion forums, but nothing seemed to work at the time. At one point I did reset the BuildConfiguration.xml to the default state (not pointing to VS 2019). I also just tried shutting everything down and restarting as a last ditch attempt... and it worked. I know that's not a very satisfying resolution to anyone that may stumble across this, though.

Thank you again, @gallonmate for your help!
